### PR TITLE
feat: add client-observed Decode Duration metric for streamed generation

### PR DESCRIPTION
**Pull request by sxuff**
## Summary

This PR adds a record-level **Decode Duration** metric for streaming inference.

Decode Duration measures the client-observed interval between the first and final non-empty streamed content responses. It complements Inter Token Latency, which normalizes the post-TTFT interval by the generated-token count. It does not claim to isolate server kernel execution time.

## Motivation

ITL answers:

> How much post-TTFT time was measured per generated-token interval?

It does not answer:

> How long was the post-TTFT stream interval observed by the client?

When output length contracts or the client undercounts generated tokens, ITL can increase sharply even when total Decode Duration is flat or lower. Reporting both metrics makes this distinction visible.

## Implementation

The new metric computes the client-observed post-TTFT interval without dividing by output-token count:

```text
Decode Duration = Request Latency - Time to First Token
```

For streamed generation, this corresponds to the interval between the first and final non-empty content responses observed by the client. Existing ITL semantics are unchanged.

## Documentation

The metrics reference now:

- defines Decode Duration as a client-observed record metric;
- distinguishes it from request latency and ITL;
- explains that ITL is generated-token-normalized;
- clarifies that Decode Duration includes response delivery and client-observed gaps rather than isolating server kernel time;
- warns that output contraction or token undercounting can increase ITL without increasing Decode Duration.

Generated documentation was refreshed through the repository hooks.

## Tests

```text
649 passed in tests/unit/metrics
pre-commit passed
ruff passed
ruff format passed
```

The added unit tests cover:

- ordinary streamed generation;
- a single output event;
- missing timing inputs;
- rejection of an invalid negative interval.

An end-to-end profile against AIPerf's mock server also exported the metric successfully:

```text
Configured mock TTFT: 20 ms
Configured mock ITL: 5 ms
Observed Decode Duration: 40.22 ms average
Observed ITL: 4.94 ms average
Records: 3
```

## Compatibility

- Existing metric names and ITL calculations are unchanged.
- The change adds a separate metric rather than redefining ITL.
- Existing consumers can ignore the new field.

## Summary by CodeRabbit

- **New Features**
  - Added the **Decode Duration** streaming metric, measuring client-observed time between the first and final streamed content responses.
  - Reported in milliseconds, with percentile support and separate error-adjusted behavior.

- **Documentation**
  - Updated the metrics reference with a full **Decode Duration** definition.
  - Refined **Inter Token Latency** notes, including updated prerequisites and clarification that ITL is token-normalized (not raw wall-clock decode time).

- **Tests**
  - Added unit tests covering Decode Duration computation, edge cases, and validation of error-adjusted percentile outputs.

### CODE REVIEW

**Review by ajcasagrande — APPROVED** [review]
LGTM

File: src/aiperf/metrics/types/decode_duration_metric.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+class DecodeDurationMetric(BaseRecordMetric[int]):
+    """Client-observed interval from the first to final content response."""
+
+    tag = "decode_duration"
+    header = "Decode Duration"
+    short_header = "Decode Duration"
+    unit = MetricTimeUnit.NANOSECONDS
+    display_unit = MetricTimeUnit.MILLISECONDS
+    display_order = 350
+    flags = MetricFlags.STREAMING_TOKENS_ONLY
```
[the remark this answers is not in the document]
Comment (ajcasagrande) [reply src/aiperf/metrics/types/decode_duration_metric.py]: hmm, interesting. i think this is wrong

[the remark this answers is not in the document]
Comment (sxuff) [reply src/aiperf/metrics/types/decode_duration_metric.py]: Yeah, I think I agree this should stay success-only.

Decode duration is the observed interval after TTFT, so for failures that do not produce a usable streamed token span, inflating it with failed requests feels less well-defined than request latency/TTFT

[the remark this answers is not in the document]
Comment (ajcasagrande) [reply src/aiperf/metrics/types/decode_duration_metric.py]: @sxuff actually it turns out the name of the flag is just bad. what it actually does is it will create an **additional** metric named `adj_decode_duration` whenever failed requests > 0. it will **never** affect the actual `decode_duration` metric which stays success-only.

I think may be worth adding to stay inline with the other ones. thoughts? See #688 for more details.

[the remark this answers is not in the document]
Comment (ajcasagrande) [reply src/aiperf/metrics/types/decode_duration_metric.py]: I am also fine with you renaming the flag across all usages if u find a better name for it.

[the remark this answers is not in the document]
Comment (sxuff) [reply src/aiperf/metrics/types/decode_duration_metric.py]: Ah got it, thanks for clarifying

Added it to `DecodeDurationMetric.flags` and included a regression test to make the semantics explicit: `decode_duration` remains success-only, while `adj_decode_duration` is emitted separately when failed requests are present.

**Review by ajcasagrande — APPROVED** [review]
LGTM. Thanks!
